src/patternfly/components/ContextSelector/context-selector.scss

```diff
@@ -118,9 +130,35 @@
     }
   }
 
-  .pf-c-context-selector__toggle-icon {
-    margin-right: var(--pf-c-context-selector__toggle-icon--MarginRight);
-    margin-left: var(--pf-c-context-selector__toggle-icon--MarginLeft);
+  &.pf-m-plain {
+    --pf-c-context-selector__toggle--PaddingRight: var(--pf-c-context-selector__toggle--m-plain--m-text--PaddingRight);
+    --pf-c-context-selector__toggle--PaddingLeft: var(--pf-c-context-selector__toggle--m-plain--m-text--PaddingLeft);
+
+    &:not(.pf-m-text) {
+      --pf-c-context-selector__toggle--PaddingRight: var(--pf-c-context-selector__toggle--m-plain--PaddingRight);
+      --pf-c-context-selector__toggle--PaddingLeft: var(--pf-c-context-selector__toggle--m-plain--PaddingLeft);
+
+      display: inline-block;
+      width: auto;
+      color: var(--pf-c-context-selector__toggle--m-plain--Color);
+
+      &.pf-m-disabled,
+      &:disabled {
+        --pf-c-context-selector__toggle--m-plain--Color: var(--pf-c-context-selector__toggle--m-plain--disabled--Color);
+      }
+    }
+
+    &::before {
+      border: 0;
+    }
+
+    &:hover,
+    &:active,
+    &.pf-m-active,
+    &:focus,
+    .pf-c-context-selector.pf-m-expanded > & {
```

Comment:
These styles do not need to apply to `.pf-m-text`. You may be able to move this entire block into the `&:not(.pf-m-text)` selector above?